server/stream.go

```diff
@@ -1147,6 +1147,12 @@ func (s *Server) checkStreamCfg(config *StreamConfig, acc *Account) (StreamConfi
 		}
 	}
 
+	for _, subj := range cfg.Subjects {
+		if SubjectsCollide(subj, JSApiPrefix) {
```

Comment:
This will help of course - we planned to add this before but with cross accounts, domains etc it’s a mine field. And of course many other subjects are problematic also. 

We should for sure also stop just “>” for example also

Reply:
I guess this sound stop > also though as it would over lap

Reply:
that was my first thought also, but with (possibly) exported JetStream services, I wanted to target the JS API  more specifically. 

Although '>' sure poses a problem every time, I can imagine a (twisted) use-case where one would want to use a stream to keep an history of `$JS.EVENT.ADVISORY.>` for instance.

Reply:
Yeah so we definitely shouldn’t stop advisories from being infested that’s a good point. 

And to be pedantic you CAN ingest > you just need to turn off ack on the stream.

Reply:
I do seem to recall some users having a stream on the API subject without acks for audit purposes.

Reply:
Pretty good point I'll take that into account and correct my code accordingly

Reply:
for the record a test exists to ensure events, etc are OK https://github.com/nats-io/nats-server/blob/cb086bcef9e5d9f051c74319ef63c2412de093e8/server/jetstream_test.go#L1085

Reply:
> I guess this should stop > also though as it would over lap

yes, it does. Included a test case for it to make that explicit